### PR TITLE
Remove dubious warning suppressions

### PR DESCRIPTION
**Pull request by basil**
See self-review for reasoning.

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@StefanSpieker 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/main/java/hudson/security/BasicAuthenticationFilter.java
Comment (basil) [inline core/src/main/java/hudson/security/BasicAuthenticationFilter.java:101]: `ACL.impersonate` seems unlikely to be a warning. Seems the author thought that this was a justification rather than a value.

File: core/src/main/java/hudson/security/FederatedLoginService.java
Comment (basil) [inline core/src/main/java/hudson/security/FederatedLoginService.java:185]: Ditto

File: core/src/main/java/hudson/security/HudsonPrivateSecurityRealm.java
Comment (basil) [inline core/src/main/java/hudson/security/HudsonPrivateSecurityRealm.java:271]: Ditto

File: core/src/main/java/jenkins/PluginSubtypeMarker.java
Comment (basil) [inline core/src/main/java/jenkins/PluginSubtypeMarker.java:57]: Whatever this is, it smells like "since Java 5". It seems like that's in the rear-view mirror now that we're on Java 8.

File: core/src/main/java/jenkins/util/TreeString.java
Comment (basil) [inline core/src/main/java/jenkins/util/TreeString.java:46]: This file complies with our Checkstyle checks, so I see no reason to turn Checkstyle off.

File: core/src/main/java/jenkins/util/TreeString.java
Comment (basil) [inline core/src/main/java/jenkins/util/TreeString.java:47]: We don't run PMD in our build, so I don't think we need to suppress PMD errors.

File: core/src/main/java/jenkins/util/TreeString.java
Comment (basil) [inline core/src/main/java/jenkins/util/TreeString.java:194]: Seemed unnecessarily broad.

File: core/src/main/java/jenkins/util/TreeStringBuilder.java
Comment (basil) [inline core/src/main/java/jenkins/util/TreeStringBuilder.java:23]: We don't run PMD in our build, so I don't think we need to suppress PMD errors.

File: core/src/main/java/jenkins/util/TreeStringBuilder.java
Comment (basil) [inline core/src/main/java/jenkins/util/TreeStringBuilder.java:24]: This file complies with our Checkstyle checks, so I see no reason to turn Checkstyle off.

File: core/src/test/java/jenkins/util/TreeStringBuilderTest.java
Comment (basil) [inline core/src/test/java/jenkins/util/TreeStringBuilderTest.java:16]: This file complies with our Checkstyle checks, so I see no reason to turn Checkstyle off.

File: core/src/test/java/jenkins/util/TreeStringBuilderTest.java
Comment (basil) [inline core/src/test/java/jenkins/util/TreeStringBuilderTest.java:15]: We don't run PMD in our build, so I don't think we need to suppress PMD errors.

**Review by StefanSpieker — APPROVED** [review]

**Review by timja — APPROVED** [review]
I suspect some of that code was copied from elsewhere with the suppressions

**Review by res0nance — APPROVED** [review]

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
